panic: interface conversion: interface {} is nil, not string [redis.go:1517]

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
JuiceFS will crash if the third redis key is missing when attempting to call doRename() on a inode, I've found this can happen with a failed upload to object storage (caused by a incorrect configuration by us). The juicefs.log will spit out the following error.

```2023/02/10 09:03:50.933179 juicefs[54762] <WARNING>: no attribute for inode 437819 (436905, thumbPreview.jpg) [redis.go:1514]
panic: interface conversion: interface {} is nil, not string

goroutine 7495667 [running]:
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).doRename.func1(0xc011c95380)
       	/root/juicefs/pkg/meta/redis.go:1517 +0x2458
github.com/go-redis/redis/v8.(*Client).Watch(0xc003a1d540?, {0x7f54d00c2040, 0xc003a1d540}, 0xc01554a360, {0xc0129f47c0, 0x4, 0x4})
       	/root/go/pkg/mod/github.com/go-redis/redis/v8@v8.11.5/tx.go:75 +0xf1
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).txn(0xc0000be140, {0x35cabe0?, 0xc003a1d540}, 0x358366a?, {0xc0129f47c0?, 0x4, 0x4})
       	/root/juicefs/pkg/meta/redis.go:775 +0x36c
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).doRename(0xc0000be140, {0x35cabe0?, 0xc003a1d540}, 0x6aaa9, {0xc011bac870, 0x27}, 0x6aaa9, {0xc0129b3590, 0x10}, 0x0, ...)
       	/root/juicefs/pkg/meta/redis.go:1433 +0x5f7
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).Rename(0xc00007f400, {0x35cabe0, 0xc003a1d540}, 0x6aaa9, {0xc011bac870, 0x27}, 0x6aaa9, {0xc0129b3590, 0x10}, 0x0, ...)
       	/root/juicefs/pkg/meta/base.go:885 +0x31d
github.com/juicedata/juicefs/pkg/vfs.(*VFS).Rename(0xc0008200b0, {0x35cb418?, 0xc003a1d540}, 0x6aaa9, {0xc011bac870, 0x27}, 0x6aaa9, {0xc0129b3590, 0x10}, 0x0)
       	/root/juicefs/pkg/vfs/vfs.go:261 +0x2be
github.com/juicedata/juicefs/pkg/fuse.(*fileSystem).Rename(0xc000c93660, 0x18?, 0xc00002df00, {0xc011bac870, 0x27}, {0xc0129b3590, 0x10})
       	/root/juicefs/pkg/fuse/fuse.go:160 +0xeb
github.com/hanwen/go-fuse/v2/fuse.doRename(0xc000be09a0, 0xc01109c900)
       	/root/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20221212033224-ed3172586347/fuse/opcode.go:435 +0xf1
github.com/hanwen/go-fuse/v2/fuse.(*Server).handleRequest(0xc000be09a0, 0xc01109c900)
       	/root/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20221212033224-ed3172586347/fuse/server.go:492 +0x1f3
github.com/hanwen/go-fuse/v2/fuse.(*Server).loop(0xc000be09a0, 0x0?)
       	/root/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20221212033224-ed3172586347/fuse/server.go:465 +0x108
created by github.com/hanwen/go-fuse/v2/fuse.(*Server).readRequest
       	/root/go/pkg/mod/github.com/juicedata/go-fuse/v2@v2.1.1-0.20221212033224-ed3172586347/fuse/server.go:332 +0x52b
```

This is because whilst the code has a check for passed interface being nil (which generates the no attribute warning), it will still go ahead and attempt to use it.

**What you expected to happen**:
JuiceFS should gracefully handle the error and fail the operation 

**How to reproduce it (as minimally and precisely as possible)**:
Our object storage was incorrectly configured that lead lead to these Redis keys not existing. Once we resolved the configuration and deployed the fix below, the issue was solved without needing to reformat by overwriting the failed files.

**Anything else we need to know?**
I was able to resolve the no attribute issue and crash by deploying this patch and overwriting the broken files.

```
diff --git a/pkg/meta/redis.go b/pkg/meta/redis.go
index 8652ad7f..14f38b70 100644
--- a/pkg/meta/redis.go
+++ b/pkg/meta/redis.go
@@ -1518,8 +1518,9 @@ func (m *redisMeta) doRename(ctx Context, parentSrc Ino, nameSrc string, parentD
                        if rs[3] == nil {
                                logger.Warnf("no attribute for inode %d (%d, %s)", dino, parentDst, nameDst)
                                trash = 0
+                       } else {
+                               m.parseAttr([]byte(rs[3].(string)), &tattr)
                        }
-                       m.parseAttr([]byte(rs[3].(string)), &tattr)
                        if (tattr.Flags&FlagAppend) != 0 || (tattr.Flags&FlagImmutable) != 0 {
                                return syscall.EPERM
                        }
```
**Environment**: Rocky Linux 8
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version: juicefs version 1.1.0-dev+2023-02-17.ccbd646e
- Cloud provider or hardware configuration running JuiceFS: OVH
- OS (e.g `cat /etc/os-release`): 8.5 (Green Obsidian)
- Kernel (e.g. `uname -a`): 4.18.0-348.23.1.el8_5.x86_64
- Object storage (cloud provider and region, or self maintained): OVH
- Metadata engine info (version, cloud provider managed or self maintained): Redis (KeyDB Flash)
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/redis.go
+++ b/pkg/meta/redis.go
@@ -1518,8 +1518,9 @@ func (m *redisMeta) doRename(ctx Context, parentSrc Ino, nameSrc string, parentD
 			if rs[3] == nil {
 				logger.Warnf("no attribute for inode %d (%d, %s)", dino, parentDst, nameDst)
 				trash = 0
+			} else {
+				m.parseAttr([]byte(rs[3].(string)), &tattr)
 			}
-			m.parseAttr([]byte(rs[3].(string)), &tattr)
 			if (tattr.Flags&FlagAppend) != 0 || (tattr.Flags&FlagImmutable) != 0 {
 				return syscall.EPERM
 			}

